how-to-guides/re-run-failed-training/notebooks/re_run_failed_training.ipynb

```diff
@@ -0,0 +1,469 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How to re-run failed training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "When you are executing a model training script that’s being tracked in Neptune and it fails in the middle, you can easily re-run it with the same metadata such as hyperparameters, data, and code version.\n",
+    "\n",
+    "By the end of this guide, you will learn how to re-open a failed Neptune run to fetch the metadata needed to re-run it and log all metadata from the model training, validation, or testing to a new run, to save results you didn’t get from the failed run.\n",
+    "\n",
+    "[See this example in Neptune](https://app.neptune.ai/common/showroom/e/SHOW-3695)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Before you start\n",
+    "\n",
+    "Make sure that you have:\n",
+    "* [Python 3.7+ installed](https://www.python.org/downloads/),\n",
+    "* [Basic familiarity with Neptune (create run and log metadata to it)](https://docs.neptune.ai/usage/#getting-started)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "! pip install -U neptune-client torch==1.10.2 torchvision==0.11.3 "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: Get Run ID\n",
+    "You will get the run ID of the failed run **programmatically**.\n",
+    "\n",
+    "**Note**: To log or retrieve metadata from Neptune, you need the project name and the API token\n",
+    "\n",
+    "To make this example easy to follow, we'll log the metadata to the public project **'common/showroom'** using a shared token for anonymous logging.\n",
+    "\n",
+    "**(Optional)** If you want to log to your own project, you need a [Neptune account](https://app.neptune.ai/register/) and a [project](https://docs.neptune.ai/setup/creating_project).\n",
+    "Then you can pass [project](https://docs.neptune.ai/setup/creating_project/#next-steps) and [api_token](https://docs.neptune.ai/setup/setting_api_token/#setting-your-api-token) arguments to the `init_run()` method.\n",
+    "\n",
+    "`run = neptune.init(api_token='YOUR_API_TOKEN', project='YOUR_WORKSPACE/YOUR_PROJECT')` \n"
```

Comment:
Passing the API token in plain text to the `api_token` argument is not a best practice, so we should not encourage it in any of our guides. See the [Notebook](https://colab.research.google.com/drive/1uQgN0uAF74cJXh3v9rmYcdOzUQjkMrIT?usp=sharing) template for how to set it with `getpass()`.